devops/providers/azure-devops/templates/infrastructure/cicd/test-harness-checkout.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+echo "Checking out test harness"
+
+CHECKOUT_DIR="./test-harness-repo-local-checkout/"
+rm -rf $CHECKOUT_DIR
+
+git clone --depth 1 $TEST_HARNESS_REPOSITORY_GIT_URL $CHECKOUT_DIR
```

Comment:
NIT -- I'm going to put this here, but looks like it applies to most of the scripts in this PR. You should put double quotes around variables to prevent reinterpretation of special characters.

```suggestion
git clone --depth 1 "$TEST_HARNESS_REPOSITORY_GIT_URL" "$CHECKOUT_DIR"
```

Reply:
I've gone ahead and updated the scripts to use quotes where appropriate